File: 프로그래머스/Summer,Winter Coding(~2018)/소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
#include <cmath>
#include <set>
using namespace std;

bool isPrime(int n){
    if(n == 1) return false;
    if(n == 2) return true;
    if(!(n%2)) return false;
    
    int s = sqrt(n);
    for(int i=3; i<=s; i+=2)
        if(!(n%i)) return false;
    
    return true;
}

int solution(vector<int> nums){
    int cnt = 0;
    for(int i=0; i<nums.size()-2; i++){
        for(int j=i+1; j<nums.size()-1; j++){
            for(int k=j+1; k<nums.size(); k++){
                int tmp = nums[i]+nums[j]+nums[k];
                if(isPrime(tmp)) cnt++;
            }
        }
    }
    
    return cnt;
}

File: 프로그래머스/Summer,Winter Coding(~2018)/소수 만들기/소수 만들기.py
from math import sqrt

def isPrime(n):
    if n == 1: return False
    if n == 2: return True
    if n%2 == 0: return False

    s = int(sqrt(n))
    for i in range(3, s+1, 2):
        if n%i == 0: return False

    return True

def solution(nums):
    cnt = 0
    for i in range(len(nums)-2):
        for j in range(i+1, len(nums)-1):
            for k in range(j+1, len(nums)):
                tmp = nums[i]+nums[j]+nums[k]
                if isPrime(tmp): cnt += 1

    return cnt
